app/jobs/spree_mailchimp_ecommerce/update_order_cart_job.rb

```diff
@@ -1,10 +1,17 @@
 module SpreeMailchimpEcommerce
   class UpdateOrderCartJob < ApplicationJob
     def perform(order_id)
-      order = Spree::Order.find(order_id)
-      return unless order.mailchimp_cart
+      order = Spree::Order.find_by(id: order_id)
+      return unless order
 
-      gibbon_store.carts(order.number).update(body: order.reload.mailchimp_cart)
+      mailchimp_cart = order.mailchimp_cart
+      return unless mailchimp_cart
+
+      begin
+        gibbon_store.carts(order.number).update(body: mailchimp_cart)
+      rescue Gibbon::MailChimpError => e
+        gibbon_store.carts.create(body: mailchimp_cart) if e.status_code == 404
```

Comment:
Is there a way to rescue specifically the 404? If not do you think it'd be good to re-raise if the error is not a 404 so that the exception get's caught on the logs?